store/iavl/store.go

```diff
@@ -80,9 +87,10 @@ func UnsafeNewStore(tree *iavl.MutableTree) *Store {
 // be used for querying and iteration only. If the version does not exist or has
 // been pruned, an error will be returned. Any mutable operations executed will
 // result in a panic.
+// TODO: CHANGE THIS DOCUMENTATION
 func (st *Store) GetImmutable(version int64) (*Store, error) {
 	if !st.VersionExists(version) {
-		return nil, iavl.ErrVersionDoesNotExist
+		return &Store{tree: &immutableTree{&iavl.ImmutableTree{}}}, nil
```

Comment:
Remind me why we're doing this?

Reply:
It was added to make sure that quering a store using a height before the upgrade (the time in which it was created) doesn't return an error. Can we remove this now or is it still necessary?

Reply:
As long as we ensure an error isn't returned and the query is handled gracefully 👍

Reply:
Ok I've tried reverted this and the `TestMultistoreLoadWithUpgrade` test is still passing so I think we can keep it as it was.

Reply:
Nevermind, this is still necessary otherwise some queries will fail